nusamai/src/transformer/transform/lods.rs

```diff
@@ -29,16 +31,68 @@ impl FilterLodTransform {
 /// Transform to filter and split the LODs
 impl Transform for FilterLodTransform {
     fn transform(&mut self, _feedback: &Feedback, mut entity: Entity, out: &mut Vec<Entity>) {
-        let lods = find_lods(&entity.root) & self.mask;
-
-        let target_lod = match self.mode {
-            LodFilterMode::Highest => lods.highest_lod(),
-            LodFilterMode::Lowest => lods.lowest_lod(),
-        };
-
-        if let Some(target_lod) = target_lod {
-            edit_tree(&mut entity.root, target_lod);
-            out.push(entity);
+        match self.mode {
+            LodFilterMode::TexturedHighest => {
+                // TODO: Processing needs to be optimised
+                let original_lods = find_lods(&entity.root) & self.mask;
```

Comment:
シンプルに、こんな感じではいかがでしょう。
一応、自分で動かしてみたところ、動作はしているように見えました。
```rust
let available_lods = find_lods(&entity.root) & self.mask;
let mut highest_textured_lod = None;

// 「最大のLOD」は初めから決まっている。テクスチャが存在しなければ、即座に最大LODを返却できる
let highest_available_lod = available_lods.highest_lod().unwrap_or(0);

// revで逆順イテレータを作成
for lod in (0..=highest_available_lod).rev() {
	if available_lods.0 & (1 << lod) != 0 {
		edit_tree(&mut entity.root, lod);

		let has_textures = {
			let appearance = entity.appearance_store.read().unwrap();
			!appearance.textures.is_empty()
		};

		// テクスチャが存在するLODが見つかれば、それを保存して終了
		if has_textures {
			highest_textured_lod = Some(lod);
			break;
		}
	}
}

// highest_textured_lodがNoneではないならhighest_textured_lodを利用
// Noneならhighest_available_lodを利用
if let Some(lod) = highest_textured_lod.or(Some(highest_available_lod)) {
	edit_tree(&mut entity.root, lod);
	out.push(entity);
}
```

```suggestion
                let original_lods = find_lods(&entity.root) & self.mask;
```

Reply:
https://github.com/MIERUNE/plateau-gis-converter/pull/666 で対応。